webapp/frontend/lib/trade/trade_screen_order_form.dart

```diff
@@ -0,0 +1,166 @@
+            onChanged: (quantity) => setState(() {
+              _quantity = Usd.parseString(quantity);
+              updateOrderValues();
+            }),
+          ),
+        ),
+        spaceBetweenRows,
+        Align(
+          alignment: AlignmentDirectional.centerEnd,
+          child: AmountInputField(
+            value: _leverage,
+            enabled: true,
+            label: "Leverage",
+            textAlign: TextAlign.right,
+            onChanged: (leverage) => setState(() {
+              _leverage = Leverage(int.parse(leverage));
+              updateOrderValues();
+            }),
+          ),
+        ),
+        spaceBetweenRows,
+        Align(
+          alignment: AlignmentDirectional.centerEnd,
+          child: AmountInputField(
+            enabled: false,
+            label: isBuy ? "Ask Price" : "Bid Price",
+            textAlign: TextAlign.right,
+            suffixIcon: const Icon(FontAwesomeIcons.dollarSign),
+            controller: _latestPriceController,
+          ),
+        ),
+        spaceBetweenRows,
+        Align(
+          alignment: AlignmentDirectional.centerEnd,
+          child: AmountInputField(
+            enabled: false,
+            label: "Margin",
+            suffixIcon: const Icon(BitcoinIcons.satoshi_v1),
+            textAlign: TextAlign.right,
+            controller: _marginController,
+          ),
+        ),
+        spaceBetweenRows,
+        Align(
+          alignment: AlignmentDirectional.centerEnd,
+          child: AmountInputField(
+            enabled: false,
+            label: "Liquidation",
+            suffixIcon: const Icon(FontAwesomeIcons.dollarSign),
+            textAlign: TextAlign.right,
+            controller: _liquidationPriceController,
+          ),
+        ),
+        spaceBetweenRows,
+        spaceBetweenRows,
+        Align(
+          alignment: AlignmentDirectional.center,
+          child: ElevatedButton(
+              onPressed: () {},
+              style: ElevatedButton.styleFrom(
+                  backgroundColor: isBuy ? buyButtonColor : sellButtonColor,
+                  minimumSize: const Size.fromHeight(50)),
+              child: isBuy ? const Text("Buy") : const Text("Sell")),
+        ),
+      ],
+    );
+  }
+
+  void updateOrderValues() {
+    if (_quantity != null && _quote != null) {
+      _marginController.text = calculateMargin(_quantity!, _quote!, _leverage, isBuy).formatted();
+      _liquidationPriceController.text =
+          calculateLiquidationPrice(_quantity!, _quote!, _leverage, widget.maintenanceMargin, isBuy)
+              .formatted();
+    }
+
+    if (isBuy && _quote != null && _quote!.ask != null) {
+      _latestPriceController.text = _quote!.ask!.formatted();
+    } else if (!isBuy && _quote != null && _quote!.bid != null) {
+      _latestPriceController.text = _quote!.bid!.formatted();
+    }
+  }
+}
+
+Amount calculateMargin(Usd quantity, Quote quote, Leverage leverage, bool isLong) {
```

Comment:
🤔 Shouldn't that logic come from the backend?

Reply:
yes, it will :) Just a placeholder calculation for now

Reply:
I went down this rabbit hole for a while but decided against it because it adds an extra round trip which may be costly.